### DISCUSSION TITLE
missav.com – VDH immediately aborts download with versions later than 8.0.0.6 (for Firefox) or 8.2.0.20 (for Chrome)

### DISCUSSION THREAD

**Discussion by HereButNotaGit** [aclap-dev/video-downloadhelper#437] [Help and Q&A]
### Your question

CAUTION: missav.com is an ADULT SITE – NOT SAFE FOR WORK.

My tests below are conducted on the most recent releases of Firefox (125.0.2) and Chromium (124.0.6367.79).  My default browser of use is Firefox, but I sometimes make use of Chromium or Chrome for reality-checking possible issues with Firefox.

VDH 8.0.0.6 (for Firefox) and VDH 8.2.0.20 (for Chrome) appear to be the most recent versions of VDH that will work with missav.com.  Attempts to download using VDH versions later than those for the respective browsers immediately abort with errors as listed in my testing results below.

I archive full offline installers for software wherever possible, so I have been able to test with many previous versions of VDH.

The homepage of missav.com presents a large random selection of "recommended" videos.  For my testing, I just picked videos at random from that homepage.

Below are the results of my testing, working backwards from the most recent version of VDH until I arrived at a version that worked.  Where I say that a test download "worked", I didn't download it to completion; I let it run long enough to decide that there was no reason to think that it wouldn't succeed.  I've in the past few days completed a succession of real-world downloads from this site using Firefox and VDH 8.0.0.6 or earlier, and it all worked like a charm.

==============================================================

TESTS USING FIREFOX 125.0.2

Firefox / 8.2.2.8

ERROR:

moz-extension://5a39fa59-d6f0-472a-827d-73ff77747e1c/background/main.js:1
receive@moz-extension://5a39fa59-d6f0-472a-827d-73ff77747e1c/background/main.js:1:13157
nd</callCatchAppNotFound/</<@moz-extension://5a39fa59-d6f0-472a-827d-73ff77747e1c/background/main.js:1:42626

Firefox / 8.2.1.1

ERROR:

moz-extension://5a39fa59-d6f0-472a-827d-73ff77747e1c/background/main.js:1
receive@moz-extension://5a39fa59-d6f0-472a-827d-73ff77747e1c/background/main.js:1:13157
Zp</callCatchAppNotFound/</<@moz-extension://5a39fa59-d6f0-472a-827d-73ff77747e1c/background/main.js:1:42626

Firefox / 8.2.0.23

ERROR:

[in#0 @ 000001b1051baa40] Error opening input: Server returned 403 Forbidden (access denied)
Error opening input file https://surrit.com/cb954fad-c8c7-494d-98b6-0666526020c0/720p/video.m3u8.
Error opening input files: Server returned 403 Forbidden (access denied)

Firefox / 8.1.4.0

ERROR:

moz-extension://5a39fa59-d6f0-472a-827d-73ff77747e1c/background/main.js:31
cn</grabInfo/<@moz-extension://5a39fa59-d6f0-472a-827d-73ff77747e1c/background/main.js:31:88385

SUCCESS:

Firefox / 8.0.0.6

Worked on 3 randomly and consecutively selected videos without problems

==============================================================

TESTS USING CHROMIUM 124.0.6367.79

Chromium / 8.2.2.9

Error: Exit code: 1

at as.receive (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:13157)
at chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:42624

Chromium / 8.2.1.1

Error: Exit code: 1

at rs.receive (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:13157)
at chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:42624

Chromium / 8.2.0.24

ERROR:

[in#0 @ 000002059ff27580] Error opening input: Server returned 403 Forbidden (access denied)
Error opening input file https://surrit.com/1b06196d-2d06-4e7e-be94-d8f7cd776eb3/1080p/video.m3u8.
Error opening input files: Server returned 403 Forbidden (access denied)

Chromium / 8.2.0.23

ERROR:

[in#0 @ 000002a1122075c0] Error opening input: Server returned 403 Forbidden (access denied)
Error opening input file https://surrit.com/ebff5549-13ea-4c30-9133-6420f45cd2c6/1280x720/video.m3u8.
Error opening input files: Server returned 403 Forbidden (access denied)

Chromium / 8.2.0.21

ERROR:

[in#0 @ 0000017ca51e7600] Error opening input: Server returned 403 Forbidden (access denied)
Error opening input file https://surrit.com/f001f86d-a001-4d51-b47c-4ac186aeaab2/1280x720/video.m3u8.
Error opening input files: Server returned 403 Forbidden (access denied)

SUCCESS:

Chromium / 8.2.0.20

Worked on 3 randomly and consecutively selected videos without problems

### Addon & CoApp version and details

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-GB
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Creator",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

version: 8.2.2.9
channel: stable
build date: 2024-03-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-GB
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Creator",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Creator",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by GrampaWildWilly** [comment]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3447, Firefox 125.0.2 64-bit, licensed VDH 8.2.2.8a1 beta, CoApp 2.0.19.  I do not work for VDH.  I'm just another user.

There have been a number of reports of problems with this site over the past few months.  In every case, I have had no problems downloading from this site.  Once again today, choosing something at random (as you did), it downloaded fine.  I think the evidence is that you need to get the latest beta.  But that may be a bit tricky.  It may still be true (I have not checked) that the link for a beta on the Wiki here is still faulty.  In that case, you must follow the instructions I posted here #391 in order to get the same beta version I have.  Do let us know how that works for you.

**Reply by HereButNotaGit** [reply]
GrampaWildWilly, thank you for your suggestions.

Firstly, I should explain that my normal method of updating Firefox extensions is to manually check daily if an extension update is available, and if so, download the .xpi installer file, and then update the extension using the Install Add-on From File option.  If I'm feeling lazy, I update over the top of the existing extension.  If I'm feeling less lazy, I remove the old extension first, but that means that I then have to re-enter my couple of customised settings later.  To ensure a maximally clean result, I shut down Firefox completely before installing the updated extension in order to flush any remnants of the removed older extension hanging around in Firefox's memory space.

I've downloaded and saved the offline installer for VDH 8.2.2.8a1 as well.

[Off-topic, but for anyone using Chrome who would like to be able to save full offline installers for extensions in the Chrome Web Store, get the easy-to-use CRX Extractor/Downloader extension available there.  It enabled me to download and install the VDH extension from the Chrome Web Store, without any logging in, back during that period when Google were demanding age-verified login in order to access the VDH page.]

Without giving you a long cock-and-bull story, I've been flipping back and forward between 8.2.2.8 and 8.2.2.8a1, swapping between the two with varying degrees of "cleanliness" according to what I described three paragraphs up from here, and I've been getting madly erratic outcomes.  For instance, at one stage I got 8.2.2.8 apparently working normally, and then when I advanced to 8.2.2.8a1, I again got the dreaded familiar error messages.  However, when I then cleared the error notifications (not the hits), I was then shown that two downloads were in progress – downloads that had supposedly been summarily aborted.  Weird!  It's like treading on eggshells.

As I write this paragraph, I'm now getting apparently normal download commencements using 8.2.2.8a1.  All I can do for the moment is leave it as it is, hope that the next update of VDH will be more stable, but keep VDH 8.0.0.6 or earlier on hand in case I need to revert to it.

**Comment by DenisXuGithub** [comment]
I maybe have the same issue on that website, i do not know that is happen?

1.. When website can't download videos by VDH, the interesting thing is that you can still play the full video on that website page, but you just cannot download the video using VDH.

2..If you are lucky enough can do download on this website now, maybe after a few many minutes, the file downloading is easily interrupted  and not the full video, however, you can still play the full video on this website page.

3..If you are lucky now, you can downloaded the 1 or 2 or 5 videos, but when you continue to downloading more video, the situation still on the table, you will face the wrong message showing up.

Remember, when you got the error message from VDH, you still can play the full video on that website page.

this picture is mean ...
! download failure
Unfortunately we are unable to download this video, please report this site so we can update it.
![VDH](https://github.com/user-attachments/assets/d3693c61-70e2-4d8b-8aba-dccae0c9ad7e)

Firefox : 129.0.2 (64bit)
VDH : 9.1.0.42 (stable)
Windows : 10 Pro (22H2)

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

Things have progressed a very great deal since this discussion was opened.  My setup now is Windows 11 Pro 64-bit version 23H2 build 22631.4169, Firefox 130.0 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20.  This is the latest of everything, latest Microsoft system maintenance for Windows 11, latest Firefox, latest VDH beta for Firefox, latest CoApp.

This web site & Pornhub exhibit the same behavior.  You start a download.  It seems to be running fine but then it seems to stop downloading.  If you wait a while -- seconds, minutes, there's no predicting -- it might resume downloading.  I'm generally not that patient.  I use the `Reload addon` button in the VDH settings to clear the download.  I reload the web page.  I do the download in VDH again.  If the same stall happens again, I go through the rigamarole again.  I do it again & again & again until it works.  Working looks like the download starts at a few million bytes per second & continues to download at a steady speed, sometimes an ever increasing speed, until the entire video completely downloads.  I have seen Pornhub download speeds approaching 30 million bytes per second (as reported in the Resource Monitor).  More often, when the download works, it seems to average something over 10 million bytes per second.  But if the download speed suddenly drops, I don't waste my time.  I reload VDH, reload the web page, & download again.  I see no point in wasting my time on a download that may or may not complete.  When I retry & I see it's downloading at a steady pace, I know it will complete.

The trick is to have patience & keep retrying.  The web sites are temperamental & inconsistent.  There is nothing VDH can do about this.  Just retry until it works.

**Reply by GrampaWildWilly** [reply]
Being able to play a video in a web page & being able to download it with VDH are 2 separate, independent things.  Being able to play the video does not predict whether VDH can download it.  The player in the web page & VDH are just different tools that use different ways of accessing the video.  Don't draw a connection between the two.

**Reply by DenisXuGithub** [reply]
1..yes, i do understand they are different method to get the video. :)

2..i have figure out a method to downloading video, and it working, and no version issue, and easy.
but i not sure if i explain it in here, the method will be block and gone.

3..i have discover the download will be easy interrupt, because the VDH is not support RESUME download ?
so, if you face the block or interrupt time, the downloading will be easy to interrupt by website or file-site.

4..that website or file-site maybe using the network traffic detect, if you have download to the unknown limit, that will be block or interrupt the few unknown minutes, and going the few unknown minutes later, you still can continue a new download with the unknown bytes, but there still can not guarantee the videos can be downloaded with complete.

thanks your information here, good luck other one. :)

**Comment by HereButNotaGit** [comment]
Thank you for your comments, GrampaWildWilly.

I'm only an occasional user of VDH, and that includes only downloading the occasional video from the Missav site these days, so I don't have much of any recent experiential sample to cite from.  Firefox is my default browser of use, and I also keep my OS (Win 11 Pro 64-bit), browsers and extensions up to date.  After trying various upgrades to VDH since I created this thread, and still experiencing problems when trying to download from Missav, if I want to download something from Missav nowadays, I just preemptively reinstall VDH 8.0.0.6 for the occasion and save myself a lot of messing around.  Apart from one occasion where I can recall a download dropping partway through for no apparent reason, I've found that VDH 8.0.0.6 simply works for me without hassle on that site.

I have a multitab bookmark on my Firefox that I reload every day to check for updates to any key apps on my PC, including VDH and the CoApp.  Where are you getting CoApp 2.0.20 from?  The Github CoApp Installation page is showing 2.0.19 as the current version, and if I download the installer .exe file anyway, the metadata in the file still identifies it as 2.0.19.

**Reply by GrampaWildWilly** [reply]
Click the `aclap-dev' link at the top of this page.  On that page, look for the CoApp link & click it.  On that page, look on the right side for a link to Releases & click that.  That is the repository of every CoApp there's ever been.  2.0.20 is at the top.  You have to click the Assets button to see what's there.  Paul told me it wasn't ready for prime time.  But I had already installed it when he told me that.  That was at least a month ago.  It seems as ready as anything else.

But I don't think it affects this issue.  Missav is just a temperamental site.  Retries are de rigueur.  If a backlevel version of VDH works for you, so be it.  Use it & be happy.
